embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionHandler.kt

```diff
@@ -587,14 +588,23 @@
         }
     }
 
+    private fun startPeriodicCaching(cacheCallback: Runnable) {
+        cacheCallback.run() // invoke the first periodic cache with 0 delay
+        schedulePeriodicCache(object : Runnable {
+            override fun run() {
+                cacheCallback.run()
+                schedulePeriodicCache(this)
+            }
+        })
+    }
+
     /**
      * It starts a background job that will schedule a callback to do periodic caching.
      */
-    private fun startPeriodicCaching(cacheCallback: Runnable) {
+    internal fun schedulePeriodicCache(cacheCallback: Runnable) {
         try {
-            scheduledFuture = this.sessionPeriodicCacheExecutorService.scheduleAtFixedRate(
+            scheduledFuture = this.sessionPeriodicCacheExecutorService.schedule(
```

Comment:
Yeah, kill all recurring jobs with fire. The ANR sampling (?) where the current run schedules the next